### ISSUE TITLE
"FormatException: Unexpected character (at character 1)\n<br />\n^\n"

### ISSUE THREAD

**Issue by holtzmichael** [ticgal/gappessentials#38]
Hello.

I have GLPI installed in Windows Server with Xampp.
I have install and activate the Gapp Essentials plugin, but when I try to log in on app, returns the error "FormatException: Unexpected character (at character 1)\n<br />\n^\n".

Maybe I am doing something wrong?

Thanks in advance

**Comment by holtzmichael** [comment]
Windows Server 2022
GLPI 10.0.3
GappEssentials 2.1.0
Apache, Mysql

**Comment by xacobofg** [comment]
Hello.
Try applying the following patch
[issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)

**Comment by holtzmichael** [comment]
> Hello. Try applying the following patch [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)

Hi xacobofg!
I applied the patch and now it´s working!
Thanks a lot for your help matte!

Have a nice day!

**Comment by Rafaeldgs** [comment]
Good evening!

I applied the same patch, but now the error is different: Parse essentials info type 'Smi' is not subtype of type "String"

**Comment by holtzmichael** [comment]
> Good evening!
> 
> I applied the same patch, but now the error is different: Parse essentials info type 'Smi' is not subtype of type "String"

Did you try to restart your webserver ?

**Comment by Rafaeldgs** [comment]
Yes, it gave the same error.
![2022-10-17_15-09](https://user-images.githubusercontent.com/88557932/196251132-848c7d3a-99f2-492c-84f8-d23e8cf536dd.jpg)

**Comment by PedroTicGal** [comment]
Hi @Rafaeldgs,

Could you provide us your selftest log? Just tap on "Send report to TICGAL team" button and send the log to us via email.

This way we could see what is happening and, if this is a Gapp or plugin error, fix it.

Don't forget to type this issue number on the email body.

Thank you.

**Comment by Rafaeldgs** [comment]
Good morning!
Of course, I just sent...

**Comment by Rafaeldgs** [comment]
What are the extensions?

**Comment by PedroTicGal** [comment]
In GLPI you can specify which file extensions can be uploaded.

Two of those file extensions you included in your GLPI config are numbers.

Example of extension: "jpg", "txt", "zip" …

**Comment by Rafaeldgs** [comment]
@PedroTicGal That's right, I removed the extensions (type of document in the manage menu) and the application worked perfectly, thank you very much for your support!

**Comment by PedroTicGal** [comment]
Glad to help.

Cheers!

**Comment by Rafaeldgs** [comment]
> Hello.
> Try applying the following patch
> [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)


And you need to apply this Patch too...

**Comment by Zukenshoes** [comment]
> > Olá. Tente aplicar o seguinte patch [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)
> 
> Olá xacobog! Apliquei o patch e agora está funcionando! Muito obrigado pela ajuda mate!
> 
> Tenha um bom dia!

Boa tarde @xacobofg!

Tem como você me explicar como você fez para aplicar o Patch?
Eu não tenho muito conhecimento na programação, mas estou enfrentando o mesmo problema que você relatou.

Obrigado

**Comment by Rafaeldgs** [comment]
> > > Olá. Tente aplicar o seguinte patch [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)
> > 
> > 
> > Olá xacobog! Apliquei o patch e agora está funcionando! Muito obrigado pela ajuda mate!
> > Tenha um bom dia!
> 
> Boa tarde @xacobofg!
> 
> Tem como você me explicar como você fez para aplicar o Patch? Eu não tenho muito conhecimento na programação, mas estou enfrentando o mesmo problema que você relatou.
> 
> Obrigado

Edit the www\glpi\marketplace\gappessentials\apirest.php file.
I leave it as pictured below.
![2022-10-19_15-52](https://user-images.githubusercontent.com/88557932/196779409-6f7e365c-1253-4798-a5ae-327999d315c2.jpg)

**Comment by Zukenshoes** [comment]
> > > > Olá. Tente aplicar o seguinte patch [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)
> > > 
> > > 
> > > Olá xacobog! Apliquei o patch e agora está funcionando! Muito obrigado pela ajuda companheiro! 
> > > Tenha um bom dia!
> > 
> > 
> > Boa tarde@xacobofg!
> > Tem como você me explicar como você fez para aplicar o Patch? Eu tenho muito conhecimento na programação.
> > Obrigado
> 
> Edite o arquivo www\glpi\marketplace\gappessentials\apirest.php. Deixo como na foto abaixo. ![2022-10-19_15-52](https://user-images.githubusercontent.com/88557932/196779409-6f7e365c-1253-4798-a5ae-327999d315c2.jpg)

Obrigado pelo retorno @Rafaeldgs !

**Comment by wangbing83** [comment]
> 你好。尝试应用以下修补程序[问题38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)



> Hello. Try applying the following patch [issue38.zip](https://github.com/ticgal/gappessentials/files/9802329/issue38.zip)

Hello, may I ask how to use the issue38.patch file? I don't know how to use it.
--------------
GLPI 10.0.3
GappEssentials 2.1.1
Xampp 
Apache mysql
